amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseRefresher.java

```diff
@@ -191,6 +191,21 @@ boolean evictLease(Lease lease)
     boolean updateLease(Lease lease)
         throws DependencyException, InvalidStateException, ProvisionedThroughputException;
 
+    /**
+     * Update application-specific fields of the given lease in DynamoDB. Does not update fields managed by the leasing
+     * library such as leaseCounter, leaseOwner, or leaseKey.
+     *
+     * @return true if update succeeded, false otherwise
+     *
+     * @throws InvalidStateException if lease table does not exist
+     * @throws ProvisionedThroughputException if DynamoDB update fails due to lack of capacity
+     * @throws DependencyException if DynamoDB update fails in an unexpected way
+     */
+    default void updateLease(Lease lease, UpdateField updateField)
```

Comment:
can we rename the api as updateLeaseWithShardDetails(which can update hashrange, childshards, parent shards, shard id....)

Reply:
Avoiding usage of shard in Lease interfaces. renaming to updateLeaseWithMetaInfo